cmd/thanos/compact.go

```diff
@@ -182,9 +182,11 @@ func runCompact(
 		level.Warn(logger).Log("msg", "Max compaction level is lower than should be", "current", maxCompactionLevel, "default", compactions.maxLevel())
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
You forgot to `defer cancel()`  and this doesn't look right, don't have a better context at this point than `context.Background` ?

Reply:
Actually, this line was some lines below. I moved it upper to pass context to `NewLeveledCompactor`.

Reply:
There is `cancel()` call at line `286` :)

Reply:
:+1: